Quick_Deploy/vLLM/README.md

```diff
@@ -0,0 +1,127 @@
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+-->
+
+
+# Deploying a vLLM model in Triton
+
+The following tutorial demonstrates how to deploy a simple
+[facebook/opt-125m](https://huggingface.co/facebook/opt-125m) model on
+Triton Inference Server using Triton's [Python backend](https://github.com/triton-inference-server/python_backend) and the
+[vLLM](https://github.com/vllm-project/vllm) library.
+
+*NOTE*: The tutorial is intended to be a reference example only. It is a work in progress with [known limitations](#limitations).
+
+
+## Step 1: Build a Custom Execution Environment with vLLM and other Dependencies
+
+Running vLLM within Triton container requires us to provide [custom execution environment](https://github.com/triton-inference-server/python_backend#creating-custom-execution-environments) with vLLM and other package dependencies. The provided script should build the package environment for you which will be used to load the model in Triton.
+
+```
+docker run --gpus all -it --rm -v ${PWD}:/work -w /work nvcr.io/nvidia/tritonserver:23.08-py3 ./gen_vllm_env.sh
+```
+
+This step might take a while to build the environment packages. Once complete, the provided sample [model_repository](model_repository) will be populated with `triton_python_backend_stub` and `vllm_env.tar.gz`.
+
+## Step 2: Set Up Triton Inference Server
+
+The structure of the sample model repository as follows after the Step 1 above:
+```
+model_repository/
+`-- vllm
+    |-- 1
+    |   `-- model.py
+    |-- config.pbtxt
+    |-- triton_python_backend_stub
+    |-- vllm_engine_args.json
+    `-- vllm_env.tar.gz
+
+```
+
+A sample model repository for deploying `facebook/opt-125m` using vLLM in Triton is included with this demo as `model_repository` directory. The content of `vllm_engine_args.json` is:
+
+```json
+{
+    "model": "facebook/opt-125m",
+    "disable_log_requests": "true"
+}
+```
+This file can be modified to provide further settings to the vLLM engine. See vLLM [AsyncEngineArgs](https://github.com/vllm-project/vllm/blob/32b6816e556f69f1672085a6267e8516bcb8e622/vllm/engine/arg_utils.py#L165) and [EngineArgs](https://github.com/vllm-project/vllm/blob/32b6816e556f69f1672085a6267e8516bcb8e622/vllm/engine/arg_utils.py#L11) for supported key-value pairs.
+
+*Note*: vLLM greedily consume upto 90% of the GPU's memory under default settings. You can provide appropriate fields like `gpu_memory_utilization` and other settings via [`vllm_engine_args.json`](model_repository/vllm/vllm_engine_args.json).
+
+Read through the documentation in [`model.py`](model_repository/vllm/1/model.py) to understand how to configure this sample for your use-case.
+
+Start the server like below:
+
+```
+docker run -p 8000:8000 -p 8001:8001 -p 8002:8002 --privileged --gpus all -it --rm --shm-size=8G --ulimit memlock=-1 --ulimit stack=67108864 -v ${PWD}/model_repository:/models nvcr.io/nvidia/tritonserver:23.08-py3 tritonserver --model-store=/models
```

Comment:
I think `--privileged` will make some users/customers uncomfortable. If not necessary for this demo, I'd remove it. I would generally never use it unless strictly required as some WAR.
```suggestion
docker run -p 8000:8000 -p 8001:8001 -p 8002:8002 --gpus all -it --rm --shm-size=8G --ulimit memlock=-1 --ulimit stack=67108864 -v ${PWD}/model_repository:/models nvcr.io/nvidia/tritonserver:23.08-py3 tritonserver --model-store=/models
```

Reply:
Tested without `--privileged`